### PR TITLE
feat(www): Pull in translated gatsby docs (ENABLE_LOCALIZATIONS env var needed)

### PR DESCRIPTION
**Pull request by tesseralis**
## Description

Pull in translated gatsby docs and display those docs on the site.

## Live Example

https://build-57f60042-fb09-471c-9527-4479f5ba2c36.gtsb.io/es/tutorial/

## Notes

* Languages are [Spanish](https://github.com/gatsbyjs/gatsby-es), [Indonesian](https://github.com/gatsbyjs/gatsby-id), [Polish](https://github.com/gatsbyjs/gatsby-pl), [Brazilian Portuguese](https://github.com/gatsbyjs/gatsby-pt-BR), and [Simplified Chinese](https://github.com/gatsbyjs/gatsby-zh-Hans). These are the languages so far that have the main tutorial series translated.
* Only tutorial pages are translated right now, since other directories have relatively imported components that cause the site to crash
* Translations are disabled by default, and can be turned on using by setting the `ENABLE_LOCALIZATIONS` variable
* Lots of follow up issues, listed below

## Follow-up Issues

* [ ] Show localized pages as "active" in the sidebar
* [ ] Support anchor links
* [ ] Replace all gatsby links with `localizedLink` (and lint against using the default gatsby link)

## Related Issues

fixes: #19353 #17759

### CODE REVIEW

File: www/gatsby-node.js [hunk 1]
```diff
@@ -475,6 +487,7 @@ exports.createPages = ({ graphql, actions, reporter }) => {
             node {
               fields {
                 slug
```
Comment (tesseralis) [inline www/gatsby-node.js]: It might end up causing confusion since here, "slug = path without locale".

File: www/i18n.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[
+  { "code": "es", "name": "Spanish" },
+  { "code": "id", "name": "Indonesian" },
+  { "code": "pl", "name": "Polish" },
+  { "code": "pt-BR", "name": "Brazilian Portuguese" },
+  { "code": "zh-Hans", "name": "Simplified Chinese" }
```
Comment (muescha) [inline www/i18n.json]:
```suggestion
  { "code": "zh-Hant", "name": "Simplified Chinese" }
```

**Comment by KyleAMathews** [comment]
Is there a preview of this we can look at?

**Comment by tesseralis** [comment]
@KyleAMathews I'm trying to build it on Gatsby Cloud but it's been finnicky, and there's currently no way to re-trigger a build AFAIK. Do we not have it set up to create a preview build on PRs?

**Comment by tesseralis** [comment]
Here's a version of it with localizations not enabled, to test that everything works as before:

https://build-62c1fb03-2fe2-4db8-8b36-3d08af0fdde2.gtsb.io/

**Comment by KyleAMathews** [comment]
It's not setup yet for preview builds. If you add your fork as a site however it'll build your branches.

File: www/gatsby-node.js (hunk 1 above)
Comment (LekoArts) [reply www/gatsby-node.js]: I guess in the interim phase we have to live with that, once we have all parts of the site we could make a `localizedSlug` or localize the `slug`?

File: www/src/components/site-metadata.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { graphql, useStaticQuery } from "gatsby"
 
 import gatsbyIcon from "../assets/gatsby-icon.png"
 
-const SiteMetadata = ({ pathname }) => {
+const SiteMetadata = ({ pathname, locale }) => {
```
[the remark this answers is not in the document]
Comment (LekoArts) [reply www/src/components/site-metadata.js]: For the current setup I don't think it'll make a difference. I would consider using Context if the component e.g. would be used in MDX (for ease of use).

**Pull request merged**